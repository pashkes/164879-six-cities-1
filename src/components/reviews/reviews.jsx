import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import compose from 'recompose/compose';

import {Operation} from "./../../reducer/data/data";
import {getComments} from "../../reducer/data/selectors";
import withLoadData from "./../../hocs/with-load-data/with-load-data";
import Review from "./../review/review.jsx";

export class Reviews extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {reviews} = this.props;
    return (
      <React.Fragment>
        <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews ? reviews.length : 0}</span></h2>
        <ul className="reviews__list" role="status" aria-live="polite">
          {reviews && reviews.map((review) => {
            return <Review {...review} key={review.id}/>;
          })}
        </ul>
      </React.Fragment>
    );
  }
}

Reviews.propTypes = {
  id: PropTypes.number.isRequired,
  reviews: PropTypes.array.isRequired,
  loadData: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  const reviewList = getComments(state, ownProps.id);
  return {
    reviews: reviewList || [],
    isLoading: Boolean(reviewList)
  };
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  loadData: () => dispatch(Operation.loadReviews(ownProps.id)),
});

const review = compose(
    connect(mapStateToProps, mapDispatchToProps),
    withLoadData
);

export default review(Reviews);
