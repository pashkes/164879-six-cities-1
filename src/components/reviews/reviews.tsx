import * as React from "react";
import {connect} from "react-redux";
import compose from 'recompose/compose';

import {Operation} from "./../../reducer/data/data";
import {getComments} from "../../reducer/data/selectors";
import withLoadData from "./../../hocs/with-load-data/with-load-data";
import Review from "./../review/review";

import {Comment} from "../../types";

interface Props {
  id: number,
  reviews: Comment[],
  isLoading: boolean,
  onLoadData: () => void,
}

export class Reviews extends React.PureComponent<Props> {
  render() {
    const {reviews} = this.props;

    return (
      <React.Fragment>
        <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length}</span></h2>
        <ul className="reviews__list" role="status" aria-live="polite">
          {reviews.map((review) => {
            return <Review {...review} key={review.id}/>;
          })}
        </ul>
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  const reviewList = getComments(state, ownProps.id);
  return {
    reviews: reviewList || [],
    isLoading: Boolean(reviewList)
  };
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  onLoadData: () => dispatch(Operation.loadReviews(ownProps.id)),
});

const enhance = compose(
    connect(mapStateToProps, mapDispatchToProps),
    withLoadData
);

export default enhance(Reviews);
