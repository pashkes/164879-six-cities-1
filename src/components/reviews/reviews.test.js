import React from "react";
import {shallow, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import {Reviews} from "./reviews.jsx";

configure({adapter: new Adapter()});

it(`list of reviews renders correctly`, () => {
  const review = shallow(<Reviews isLoading={false} loadData={jest.fn()} reviews={[]} id={0} />);
  expect(review).toMatchSnapshot();
});
