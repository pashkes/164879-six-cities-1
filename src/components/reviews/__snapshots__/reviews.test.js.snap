// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list of reviews renders correctly 1`] = `
<Fragment>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      0
    </span>
  </h2>
  <ul
    aria-live="polite"
    className="reviews__list"
    role="status"
  />
</Fragment>
`;
